tests/test_filesystem.py

```diff
@@ -0,0 +1,54 @@
+import os
+
+import boto3
+import fsspec
+import pytest
+from moto import mock_s3
+
+from datasets.filesystems import S3FileSystem, extract_path_from_uri, is_remote_filesystem
+
+
+@pytest.fixture(scope="function")
+def aws_credentials():
+    """Mocked AWS Credentials for moto."""
+    os.environ["AWS_ACCESS_KEY_ID"] = "fake_access_key"
+    os.environ["AWS_SECRET_ACCESS_KEY"] = "fake_secret_key"
+    os.environ["AWS_SECURITY_TOKEN"] = "fake_secrurity_token"
+    os.environ["AWS_SESSION_TOKEN"] = "fake_session_token"
+
+
+@pytest.fixture(scope="function")
+def s3(aws_credentials):
+    with mock_s3():
+        yield boto3.client("s3", region_name="us-east-1")
+
+
+@mock_s3
```

Comment:
If I understand correctly you already mock s3 in the `s3` fixture right ?
So maybe you don't need to add `@mock_s3` here ?

Reply:
True.